src/vs/editor/contrib/folding/folding.ts

```diff
@@ -228,9 +228,15 @@ export class FoldingController implements IEditorContribution {
 		if (this.rangeProvider) {
 			return this.rangeProvider;
 		}
-		this.rangeProvider = new IndentRangeProvider(editorModel); // fallback
```

Comment:
Be carefull here what to change. The goal is to
avoid flickering when folding is requests while the extension host is not yet up.

I believe the only change necessary is  in the SyntaxRangeProvider

Reply:
I don't understand why I should change `SyntaxRangeProvider` since we only need to provide it the corrects `FoldingRangeProvider`

Reply:
Ohhh, it's to avoid a race condition... I will fix that.